SystemManager/Src/SystemManager.cpp

```diff
@@ -10,37 +10,56 @@
 #include "drivers_config.hpp"
 #include "independent_watchdog.h"
 #include "tim.h"
+#include "CommonDataTypes.hpp"
+#include "config.hpp"
+#include "cmsis_os.h"
 
 SystemManager::SystemManager():
     rcController_(&huart2),
     throttleMotorChannel_(&htim2, TIM_CHANNEL_1),
     yawMotorChannel_(&htim2, TIM_CHANNEL_2),
     rollMotorChannel_(&htim2, TIM_CHANNEL_3),
     pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
-    invertedRollMotorChannel_(&htim3, TIM_CHANNEL_1),
-    watchdog_(&hiwdg)
-{}
+    watchdog_(&hiwdg),
+    am_instance_(config::flightmodes[config::DEFAULT_FLIGHTMODE].flightmodeConstructor())
```

Comment:
I think we should probably construct the flight modes all at once in one place, perhaps in the attitude manager constructor?
I actually don't know if we want to have multiple flight modes all allocated at the same time and just move a pointer to the active one or if we want to create / destroy them as needed. I think this is something that needs to be discussed, but the way you have it now is fine until we need to make a decision on this.

Reply:
Decided to construct flight mode objects on the fly for now.